lib/compat/wordpress-6.7/rest-api.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * PHP and WordPress configuration compatibility functions for the Gutenberg
+ * editor plugin changes related to REST API.
+ *
+ * @package gutenberg
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	die( 'Silence is golden.' );
+}
+
+if ( ! function_exists( 'gutenberg_register_wp_rest_post_types_meta_fields' ) ) {
+	/**
+	 * Adds `template` and `template_lock` fields to WP_REST_Post_Types_Controller class.
+	 */
+	function gutenberg_register_wp_rest_post_types_meta_fields() {
+		register_rest_field(
```

Comment:
Maybe we don't need to even expose a new field in this endpoint. It seems we only need the `keys` that have been registered. We could potentially get that information from any post, page, CPT. It seems that even if there is no default value, the custom field appears in the API with an empty value. Maybe we could fetch any post/page/CPT, and get the keys (without the values) from there.

Reply:
The key is a great approach for layout building inside templates. The default value may be necessary for post / page / cpts.

But also, it can cause the need of deleting them when editing the field if we don't use them as just a placeholder.
Maybe use a placeholder, and save as real data when the user saves the post if the field has not been edited.